prov/rxm/src/rxm_cq.c

```diff
@@ -225,6 +168,143 @@ static int rxm_match_rma_iov(struct rxm_recv_entry *recv_entry,
 	return FI_SUCCESS;
 }
 
+static int rxm_finish_recv(struct rxm_rx_buf *rx_buf)
+{
+	int ret;
+
+	if (rx_buf->recv_entry->flags & FI_COMPLETION) {
+		FI_DBG(&rxm_prov, FI_LOG_CQ, "writing recv completion\n");
+		ret = ofi_cq_write(rx_buf->ep->util_ep.rx_cq,
+				   rx_buf->recv_entry->context,
+				   rx_buf->recv_entry->comp_flags |
+				   (rx_buf->recv_entry->multi_recv_entry ?
+				    FI_MULTI_RECV : 0) |
+				   ((rx_buf->pkt.hdr.flags & OFI_REMOTE_CQ_DATA) ?
+				    FI_REMOTE_CQ_DATA : 0),
+				   rx_buf->pkt.hdr.size, NULL,
```

Comment:
The buf arg should point to the starting location of the recv buffer.

Reply:
Done